### DISCUSSION TITLE
VOLT: Why is the data passed to with() not updating on each rendering?

### DISCUSSION THREAD

**Discussion by f-ricci** [livewire/livewire#7306] [1. Help]
Hi,

I have a large list of items and I don't like the idea of using that and passing around all that metadata for on each requests. Using the classic class based components I was able to put such things in the render function and not have to deal with extra meta data going around.

Now, I can use `state` which executes every time and adds extra metadata to all requests or `with` which gets invoked only during the first time.

Am I getting something wrong or is this the way it is intended to work?

Also, why is `rendering` function not part of the hooks?

**Comment by joshhanley** [comment]
@f-ricci please ask volt questions on the volt repo 🙂 but you could use a computed property instead, which I would recommend over passing data into the render method in normal Livewire anyway.

Hope this helps!
